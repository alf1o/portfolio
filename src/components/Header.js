import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';

/**
  Render the main header.
  The menu icon is used to open the `SideMenu`.
*/
function Header({ onMenuClick }) {
  return (
    <AppBar
      position="static"
    >
      <Toolbar>
        <IconButton
          onClick={onMenuClick}
          aria-label="Menu"
        >
          <MenuIcon
            color="secondary"
          />
        </IconButton>
        <Typography
          variant="title"
        >
          Alfio Parisi
        </Typography>
      </Toolbar>
    </AppBar>
  );
}

Header.propTypes = {
  onMenuClick: PropTypes.func.isRequired
};

export default Header;
