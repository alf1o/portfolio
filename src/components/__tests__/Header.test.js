import React from 'react';
import Header from '../Header';
import { createShallow } from '@material-ui/core/test-utils';
import IconButton from '@material-ui/core/IconButton';

describe('`Header`', () => {
  let props = {};
  let mountedHeader;
  const shallow = createShallow();

  function header() {
    if (!mountedHeader) mountedHeader = shallow(<Header {...props} />);
    return mountedHeader;
  }

  beforeEach(() => {
    props = {
      onMenuClick: jest.fn()
    };
    mountedHeader = undefined;
  });

  it('should pass `onMenuClick` to `IconButton`', () => {
    expect(header().find(IconButton).props().onClick).toEqual(props.onMenuClick);
  });

  it('should match snapshot', () => {
    expect(header()).toMatchSnapshot();
  });
});
