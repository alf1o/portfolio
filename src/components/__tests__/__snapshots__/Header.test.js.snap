// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Header\` should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    onMenuClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Toolbar)>
        <WithStyles(IconButton)
                aria-label="Menu"
                onClick={[Function]}
        >
                <pure(Menu)
                        color="secondary"
                />
        </WithStyles(IconButton)>
        <WithStyles(Typography)
                color="secondary"
                variant="title"
        >
                Alfio Parisi
        </WithStyles(Typography)>
</WithStyles(Toolbar)>,
      "position": "static",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(IconButton)
            aria-label="Menu"
            onClick={[Function]}
>
            <pure(Menu)
                        color="secondary"
            />
</WithStyles(IconButton)>,
          <WithStyles(Typography)
            color="secondary"
            variant="title"
>
            Alfio Parisi
</WithStyles(Typography)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-label": "Menu",
            "children": <pure(Menu)
              color="secondary"
/>,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": "secondary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Alfio Parisi",
            "color": "secondary",
            "variant": "title",
          },
          "ref": null,
          "rendered": "Alfio Parisi",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Toolbar)>
          <WithStyles(IconButton)
                    aria-label="Menu"
                    onClick={[Function]}
          >
                    <pure(Menu)
                              color="secondary"
                    />
          </WithStyles(IconButton)>
          <WithStyles(Typography)
                    color="secondary"
                    variant="title"
          >
                    Alfio Parisi
          </WithStyles(Typography)>
</WithStyles(Toolbar)>,
        "position": "static",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(IconButton)
              aria-label="Menu"
              onClick={[Function]}
>
              <pure(Menu)
                            color="secondary"
              />
</WithStyles(IconButton)>,
            <WithStyles(Typography)
              color="secondary"
              variant="title"
>
              Alfio Parisi
</WithStyles(Typography)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Menu",
              "children": <pure(Menu)
                color="secondary"
/>,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "secondary",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Alfio Parisi",
              "color": "secondary",
              "variant": "title",
            },
            "ref": null,
            "rendered": "Alfio Parisi",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
